Opentrons/opentrons
# feat(update-server): Set buildroot hostname

Add the ability to set the device name with POST /server/name.

Also sets the correct hostname on boot.

The name endpoint is POST /server/name with a json payload of {"name": "string"}. The name has basically no restrictions other than valid utf-8, and will be truncated and transformed to form the system hostname. Right now, that system hostname is what's used by avahi to advertise, since that's the only thing you can do from a static service configuration.

Closes #3319 
Requires https://github.com/Opentrons/buildroot/pull/19
## Testing

- Install on a buildroot robot (there's no buildroot changes so `make push-buildroot` is fine
- Change the name with `POST /server/name`
- Note that it's ok, responds with the new name, and the name changes in `dns-sd` and the discovery client
- Reboot the robot
- Note that it still has the name when it boots up again
